initial blog: Paris Dev Sprint 2023 (#167)
scikit-learn/blog: _posts/2023-07-31-paris-dev-sprint.md

---
title: "scikit-learn 2023 In-person Developer Sprint in Paris, France"
date: July 31, 2023

categories:
  - Events
tags:
  - Sprints
  - Community

postauthors:
  - name: Reshama Shaikh
    website: https://reshamas.github.io
    image: reshama_shaikh.jpeg 
  - name: François Goupil
    email: francois.goupil@inria.fr
    website: https://github.com/francoisgoupil
    image: "francois_goupil.jpeg"
---
<div>
  {% include postauthor.html %}
</div>

In June of 2023 (week of June 19 to 23), the scikit-learn team held its first developers sprint since 2019! The sprint took place in Paris, France at the Dataiku office. The sprint event was an in-person event only and had 29 participants.

The following [scikit-learn team members](https://scikit-learn.org/stable/about.html) joined the sprint:  

1. Gael Varoquaux (@GaelVaroquaux)
1. Olivier Grisel (@ogrisel)
1. Guillaume Lemaitre (@glemaitre)

> 1. Tim Head (@betatim)

Reviewer: ```suggestion
1. Tim Head (@betatim)
1. Jérémie du Boisberranger (@jeremiedbb)
1. Frank Charras (@fcharras)
```

Author: @jjerphan Is Frank Charras a scikit-learn team member?

Reviewer: Franck is working on GPU implementations with Olivier and Tim.

He should be part of scikit-learn team if he's not IMO.

1. Maren Westermann
1. Arturo Amor Quiroz
1. Adrin Jalali
1. Loïc Estève
1. Thomas Fan
1. Julien Jerphanion (@jjerphan)
1. François Goupil (@francoisgoupil)
1. Roman Yurchak
1. Joris Van den Bossche

The following community members joined the sprint:  

1. Ines (intern at Dataiku)
1. Chaine San Buenaventura
1. Yann Lechelle
1. Vincent Maladière
1. Franck Charras
1. Jovan Stojanovic
1. Leo Dreyfus-Schmidt
1. Alexandre Landeau
1. Louis Fouquet
1. Alexandre Vigny
1. Harizo Rajaona
1. Samuel Ronsin
1. Riccardo Cappuzzo
1. Camille Troillard

> 1. Denis Engemann

Reviewer: ```suggestion
1. Denis Engemann
1. Léo Grinsztajn
1. Lilian Boulard
```

<figure>
 <img src="/assets/images/posts_images/2023-paris-sprint/paris_2023.jpg" alt="group of people who participated in the sprint" max-width="20%" max-height="20%" /> 
 <figcaption>

>  scikit-learn Developer Sprint, Paris, June 2023; Photo credit: <a href=" "> Inria / Photo B. Fourrier, June 2023</a>; (from left to right, back to front): XXX, XXX, François Goupil, Tim Head, Guillaume Lemaitre, XXX, XXX, Gael Varoquaux, Olivier Grisel, Alexandre Gramfort [add other people in]

Reviewer: ```suggestion
 scikit-learn Developer Sprint, Paris, June 2023; Photo credit: <a href=" "> Inria / Photo B. Fourrier, June 2023</a>; (from left to right, back to front):
Last Row: Denis Engemann, Riccardo Cappuzzo, François Goupil, Tim Head, Guillaume Lemaitre, Louis Fouquet, Jérémie du Boisberranger, Frank Charras, Léo Grinsztajn, Arturo Amor Quiroz. Middle Row: Thomas Fan, Lilian Boulard, Gaël Varoquaux,  Ines, Jovan Stojanovic, Chaine San Buenaventura. First Row: Olivier Grisel, Harizo Rajaona, Vincent Maladière.
```

 </figcaption>
</figure>

## Sponsors
- Dataiku provided the space and some of the food, as well as all of the coffee.

> - The scikit-learn consortium paid for the lunch, the travel and accommodation expenses.

Reviewer 2: I would mention that the scikit-learn consortium organized the sprint and paid for the lunch...

## Topics covered at the sprint
- PR #13649: [Monotonic constraints for Tree-based models](https://github.com/scikit-learn/scikit-learn/pull/13649) 
- Discussed the vision/future directions for the project. What is important to keep the project relevant in the future.
- Should we share some points beyond the vision statement?
- Thomas F will try and create a vision statement
- Discussed what people are keeping an eye on with a two year time scale in mind in terms of technology and developments that are relevant.
- Tim: keep improving our documentation (not just expanding it but also “gardening” to keep it readable)
- Tim: increase active outreach and communication about new features/improvements and other changes. A lot of cool things in scikit-learn are virtually unknown to the wider public (e.g. Hist grad boosting being on par with lightgbm in terms of performance, …)

### What is next?

> We are discussing co-locating with OpenML next Sep to Dec 2024 in Berlin to organize another developers' sprint. 

Reviewer 2: Next spring

Author: changing to "2024" in general.  
"spring" is different for different hemispheres, so will keep it general.

---
title: "scikit-learn 2023 In-person Developer Sprint in Paris, France"
date: July 31, 2023

categories:
  - Events
tags:
  - Sprints
  - Community

postauthors:
  - name: Reshama Shaikh
    website: https://reshamas.github.io
    image: reshama_shaikh.jpeg 
  - name: François Goupil
    email: francois.goupil@inria.fr
    website: https://github.com/francoisgoupil
    image: "francois_goupil.jpeg"
---
<div>
  {% include postauthor.html %}
</div>

> In June of 2023 (week of June 19 to 23), the scikit-learn held its first Developers sprint since 2019!  The sprint took place in Paris, France at the Dataiku office. The sprint event was an in-person event only and had 29 participants.

Author: ```suggestion
In June of 2023 (week of June 19 to 23), the scikit-learn team held its first Developers sprint since 2019!  The sprint took place in Paris, France at the Dataiku office. The sprint event was an in-person event only and had 29 participants.
```

> In June of 2023 (week of June 19 to 23), the scikit-learn held their first Developers sprint since 2019!  The sprint took place in Paris, France at the Dataiku office. The sprint event was an in-person event only and had 29 participants.

Reviewer 3: ```suggestion
In June of 2023 (week of June 19 to 23), the scikit-learn team held their first developers sprint since 2019!  The sprint took place in Paris, France at the Dataiku office. The sprint event was an in-person event only and had 29 participants.
```

Feels like there is a word missing. Could also be "... , scikit-learn held its first developers sprint".

Author: Thanks @betatim

> In June of 2023 (week of June 19 to 23), the scikit-learn held their first Developers sprint since 2019!  The sprint took place in Paris, France at the Dataiku office. The sprint event was an in-person event only and had 29 participants.

Author: ```suggestion
In June of 2023 (week of June 19 to 23), the scikit-learn held its first Developers sprint since 2019!  The sprint took place in Paris, France at the Dataiku office. The sprint event was an in-person event only and had 29 participants.
```

> In June of 2023 (week of June 19 to 23), the scikit-learn team held its first Developers sprint since 2019!  The sprint took place in Paris, France at the Dataiku office. The sprint event was an in-person event only and had 29 participants.

Author: ```suggestion
In June of 2023 (week of June 19 to 23), the scikit-learn team held its first developers sprint since 2019!  The sprint took place in Paris, France at the Dataiku office. The sprint event was an in-person event only and had 29 participants.
```

> In June of 2023 (week of June 19 to 23), the scikit-learn team held its first developers sprint since 2019!  The sprint took place in Paris, France at the Dataiku office. The sprint event was an in-person event only and had 29 participants.

Reviewer 4: I think there are two white spaces after `2019!`. :)

The following [scikit-learn team members](https://scikit-learn.org/stable/about.html) joined the sprint:  

1. Gael Varoquaux (add GitHub handle here too)
1. Olivier Grisel

> 1. Guillaume LeMaitre

Reviewer 5: ;)
```suggestion
1. Guillaume Lemaitre
```

1. Tim Head (@betatim)
1. Maren Westermann
1. Arturo Amor Quiroz
1. Adrin Jalali
1. Loïc Estève
1. Thomas Fan
1. Julien Jerphanion (@jjerphan)

> 1. Tim Head

Reviewer 4: Tim is on the list twice.

1. François Goupil (@francoisgoupil)
1. Roman Yurchak
1. Joris Van den Bossche

The following community members joined the sprint:  

> 1. Ines intern @Dataiku

Author: ```suggestion
1. Ines (intern at Dataiku)
```

1. Chaine San Buenaventura
1. Yann Lechelle
1. Vincent Maladière
1. Franck Charras
1. Jovan Stojanovic
1. Leo Dreyfus-Schmidt
1. Alexandre Landeau
1. Louis Fouquet
1. Alexandre Vigny
1. Harizo Rajaona
1. Samuel Ronsin
1. Riccardo Cappuzzo
1. Camille Troillard
1. Denis Engemann

<figure>
 <img src="/assets/images/posts_images/2023-paris-sprint/paris_2023.jpg" alt="group of people who participated in the sprint" max-width="20%" max-height="20%" /> 
 <figcaption>

>  scikit-learn Developer Sprint, Paris, June 2023; Photo credit: <a href=" "> Inria / Photo B. Fourrier, June 2023</a>; (from left to right): Gael Varoquaux, Olivier Grisel, Alexandre Gramfort [add other people in]

Reviewer 3: ```suggestion
 scikit-learn Developer Sprint, Paris, June 2023; Photo credit: <a href=" "> Inria / Photo B. Fourrier, June 2023</a>; (from left to right, back to front): XXX, XXX, François Goupil, Tim Head, Guillaume Lemaitre, XXX, XXX, Gael Varoquaux, Olivier Grisel, Alexandre Gramfort [add other people in]
```

1. Ines (intern at Dataiku)
1. Chaine San Buenaventura
1. Yann Lechelle
1. Vincent Maladière
1. Franck Charras
1. Jovan Stojanovic
1. Leo Dreyfus-Schmidt
1. Alexandre Landeau
1. Louis Fouquet
1. Alexandre Vigny
1. Harizo Rajaona
1. Samuel Ronsin
1. Riccardo Cappuzzo
1. Camille Troillard
1. Denis Engemann
1. Léo Grinsztajn
1. Lilian Boulard

<figure>
 <img src="/assets/images/posts_images/2023-paris-sprint/paris_2023.jpg" alt="group of people who participated in the sprint" max-width="20%" max-height="20%" /> 
 <figcaption>
 scikit-learn Developer Sprint, Paris, June 2023; Photo credit: <a href=" "> Inria / Photo B. Fourrier, June 2023</a>; (from left to right, back to front):
Last Row: Denis Engemann, Riccardo Cappuzzo, François Goupil, Tim Head, Guillaume Lemaitre, Louis Fouquet, Jérémie du Boisberranger, Frank Charras, Léo Grinsztajn, Arturo Amor Quiroz.
Middle Row: Thomas Fan, Lilian Boulard, Gaël Varoquaux,  Ines, Jovan Stojanovic, Chaine San Buenaventura.
First Row: Olivier Grisel, Harizo Rajaona, Vincent Maladière.
 </figcaption>
</figure>

## Sponsors
- Dataiku provided the space and some of the food, as well as all of the coffee.

> - The scikit-learn consortium paid for the lunch, the travel and accommodation expenses.

Author: ```suggestion
- The scikit-learn consortium organized the sprint, paid for the lunch, the travel and accommodation expenses.
```
